analyses/molecular-subtyping-HGG/00-HGG-select-pathology-dx.Rmd

```diff
@@ -0,0 +1,185 @@
+
+1. A defining lesion was identified in the SNV consensus file (H3 K28M or G35R/V)
+2. The `short_histology` was `HGAT`.
+
+In an upcoming release, `integrated_diagnosis`, which can be updated as the result of subtyping, will be used to populate the `short_histology` column (see [#748](https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/748)).
+Thus, molecular subtyping modules need to be upstream of `short_histology` and use the `pathology_diagnosis` and `pathology_free_text_diagnosis` fields.
+This change for this module is tracked in [#754](https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/754).
+
+Filtering on the basis of `short_histology == HGAT` is more straightforward than using the pathology diagnosis fields, so we include this notebook to put together the terms in `pathology_diagnosis` and `pathology_free_text_diagnosis`.
+
+We will use the 2016 WHO Classification as our guide ([Louis et al. _Acta Neuropathol._ doi: 10.1007/s00401-016-1545-1](10.1007/s00401-016-1545-1)) and take a look at the current version of the histology file (`release-v17-20200908`).
+
+## Set up
+
+```{r}
+library(tidyverse)
+```
+
+### Directories and files
+
+We're going to tie this to a specific release.
+
+```{r}
+data_dir <- file.path("..", "..", "data", "release-v17-20200908")
+histologies_file <- file.path(data_dir, "pbta-histologies.tsv")
+```
+
+We're going to save the pathology diagnosis information we'll use to generate the subset files in a directory `hgg-subset`.
+
+```{r}
+output_dir <- "hgg-subset"
+output_file <- file.path(output_dir, 
+                         "hgg_subtyping_path_dx_strings.json")
+```
+
+## Read in data
+
+```{r}
+histologies_df <- read_tsv(histologies_file)
+```
+
+## Explore the pathology diagnoses
+
+### `short_histology == HGAT`
+
+In the current histologies file, if we filter based on `short_histology` as we did originally, what is in the pathology diagnosis fields?
+Note that some of these values will have been altered based on earlier subtyping efforts.
+(That's why we're doing this!)
+
+```{r}
+histologies_df %>% 
+  filter(short_histology == "HGAT") %>%
+  group_by(pathology_diagnosis) %>%
+  tally() %>%
+  arrange(desc(n))
+```
+
+For the most part, this is as we would expect given the 2016 WHO classifications. 
+In an initial round of subtyping, PNET specimens were reclassified (see [this comment on #609](https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/609#issuecomment-602821376)).
+
+Let's take a look at the free text field.
+
+```{r}
+histologies_df %>% 
+  filter(short_histology == "HGAT") %>%
+  group_by(pathology_free_text_diagnosis) %>%
+  tally() %>%
+  arrange(desc(n))
+```
+
+As we might expect from a free text field, this is a lot less uniform.
+
+## Pathology diagnosis strings for inclusion
+
+These are the terms that we'll collapse together with `|` to detect strings in the `pathology_diagnosis` column.
```

Comment:
It's not a defined vocabulary as far as I can tell (this is my assumption based on the presence of both `Brainstem glioma- Diffuse intrinsic pontine glioma` and `Infiltrating Dipg` which may be incorrect), which is why I hesitate to use exact matching. But you are right that it could be more robust if made case insensitive. I'll go that route unless @jharenza replies and tells us that the values for `pathology_diagnosis` are a controlled vocabulary.

Reply:
> Can you say more about why some of the terms are excluded? For example: `High Grade Glial Neoplasma`. I see that PNET samples were reclassified later, but are there PNET samples that were not? If we include them here are we getting false positives?

Regarding the PNET samples, there's nothing in the strings used for matching that is designed to capture these PNET. The PNET samples that were reclassified should be captured by the defining lesions step in `02`.  My (non-expert) understanding is that the tumors designated as PNET are more appropriate subtyped in the non-ATRT/non-MB embryonal module. I can add text at line 72 to make excluding PNETs more clear. 

You're right that `High Grade Glial Neoplasma` should probably be included. I will make that change.

Reply:
>My (non-expert) understanding is that the tumors designated as PNET are more appropriately subtyped in the non-ATRT/non-MB embryonal module. I can add text at line 72 to make excluding PNETs more clear.


Just catching up on this, but yes, @jaclyn-taroni you are correct here.